scipy/optimize/_qap_umeyama.py

```diff
@@ -0,0 +1,17 @@
+import numpy as np
+from . import linear_sum_assignment, OptimizeResult
+
+
+def eigenvectors(x):
+    # get eigenvectors sorted by eigenvalue magnitude (large to small)
+    l, v = np.linalg.eig(x)
+    indices = np.argsort(l, kind='merge')[::-1]
```

Comment:
is X always hermitian? otherwise you can't sort reliably. If it is, then use `eigh` from scipy which auto-sorts them.

Also why `merge`?

Reply:
Thanks for pointing that daft mistake out. The matrices are not in general hermitian.

I have added handling and tests for non-hermitian matrices. The `eigenvectors` function has been replaced with plain old `eigh`.